rmw_fastrtps_dynamic_cpp/src/publisher.cpp

```diff
@@ -104,14 +104,23 @@ rmw_fastrtps_dynamic_cpp::create_publisher(
 
   /////
   // Get RMW Type Support
+  auto type_hash = rosidl_get_zero_initialized_type_hash();
```

Comment:
```suggestion
  rosidl_type_hash_t type_hash = rosidl_get_zero_initialized_type_hash();
```

(also, you need to `#include "rosidl_runtime_c/type_hash.h"` at the top of this file.